tests/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+include(CTest)
+
+set(test_script ${CMAKE_CURRENT_LIST_DIR}/run_test.py)
+
+file(GLOB files ${CMAKE_CURRENT_LIST_DIR}/tests/*)
```

Comment:
I would explicitly list the test files. Globbing files is convenient, but can match files that you don't want to get matched (here it will get everything in `tests/`) and also does not track dependencies correctly.

Reply:
Will do.

What do you mean by "does not track dependencies properly"?

Reply:
If you add new tests and issue `make test` the new tests will not be included since the build system will not know  when to call cmake to regenerate/reconfigure. However, if you explicitly add the test to CMakeLists.txt they will be included.